cmd/cluster-init/generate_job.go

```diff
@@ -0,0 +1,231 @@
+	prowconfig "k8s.io/test-infra/prow/config"
+)
+
+//TODO: not sure some of these should be constants, what if we only use them once???
+const (
+	Openshift      = "openshift"
+	Release        = "release"
+	Master         = "master"
+	ApplyConfig    = "applyconfig"
+	Latest         = "latest"
+	ConfigDir      = "config-dir"
+	Clusters       = "clusters"
+	Confirm        = "confirm"
+	As             = "as"
+	Always         = "Always"
+	Cpu            = "cpu"
+	Kubernetes     = "kubernetes"
+	AppDotCi       = "app.ci"
+	Periodic       = "periodic"
+	Apply          = "apply"
+	Ci             = "ci"
+	Io             = "io"
+	Role           = "role"
+	Infra          = "infra"
+	Registry       = "registry"
+	Org            = "org"
+	Branch         = "branch"
+	Dry            = "dry"
+	BuildFarm      = "build-farm"
+	Pull           = "pull"
+	CanBeRehearsed = "can-be-rehearsed"
+	Tmp            = "tmp"
+)
+
+func generatePeriodic(clusterName string, buildFarmDir string) prowconfig.Periodic {
+	utilityConfig := generateUtilityConfig()
+	utilityConfig.ExtraRefs = append(utilityConfig.ExtraRefs, generateReleaseRef())
+	image := fmt.Sprintf("%s:%s", ApplyConfig, Latest)
+	args := generateArgs(buildFarmDir)
+	args = append(args, fmt.Sprintf("--%s=true", Confirm))
+	container := generateContainer(image, args, generateVolumeMounts())
+	volume := generateSecretVolume(clusterName)
+	podSpec := generatePodSpec(container, []v1.Volume{volume})
+	name := fmt.Sprintf("%s-%s-%s-%s-%s-%s", Periodic, Openshift, Release, Master, clusterName, Apply)
+	jobBase := generateJobBase(podSpec, utilityConfig, name)
+	jobBase.Labels = map[string]string{
+		fmt.Sprintf("%s.%s.%s/%s", Ci, Openshift, Io, Role): Infra,
+	}
+	return prowconfig.Periodic{
+		JobBase:  jobBase,
+		Interval: "12h",
+	}
+}
+
+func generatePostsubmit(clusterName string, buildFarmDir string) prowconfig.Postsubmit {
+	utilityConfig := generateUtilityConfig()
+	image := fmt.Sprintf("%s.%s.%s.%s/%s/%s:%s", Registry, Ci, Openshift, Org, Ci, ApplyConfig, Latest)
+	args := generateArgs(buildFarmDir)
+	args = append(args, fmt.Sprintf("--%s=true", Confirm))
+	container := generateContainer(image, args, generateVolumeMounts())
+	volume := generateSecretVolume(clusterName)
+	podSpec := generatePodSpec(container, []v1.Volume{volume})
+	name := fmt.Sprintf("%s-%s-%s-%s-%s-%s-%s", Branch, Ci, Openshift, Release, Master, clusterName, Apply)
+	jobBase := generateJobBase(podSpec, utilityConfig, name)
+	jobBase.MaxConcurrency = 1
+	jobBase.Labels = map[string]string{
+		fmt.Sprintf("%s.%s.%s/%s", Ci, Openshift, Io, Role): Infra,
+	}
+	brancher := prowconfig.Brancher{
+		Branches: []string{"master"},
+	}
+	return prowconfig.Postsubmit{
+		JobBase:  jobBase,
+		Brancher: brancher,
+	}
+}
+
+func generatePresubmit(clusterName string, buildFarmDir string) prowconfig.Presubmit {
+	utilityConfig := generateUtilityConfig()
+	image := fmt.Sprintf("%s.%s.%s.%s/%s/%s:%s", Registry, Ci, Openshift, Org, Ci, ApplyConfig, Latest)
```

Comment:
You likely want the image itself to be the constant - will any individual field in this ever change?

Reply:
That makes sense, I increasingly was disliking the way I was handling the constants as they grew larger.